protocols/dcutr/src/handler/relayed.rs

```diff
@@ -99,21 +150,7 @@ impl Handler {
         >,
     ) {
         match output {
-            future::Either::Left(stream) => {
-                if self
-                    .inbound_stream
-                    .try_push(inbound::handshake(
-                        stream,
-                        self.holepunch_candidates.clone(),
-                    ))
-                    .is_err()
-                {
-                    tracing::warn!(
-                        "New inbound connect stream while still upgrading previous one. Replacing previous with new.",
-                    );
-                }
-                self.attempts += 1;
-            }
+            future::Either::Left(stream) => self.set_stream(StreamType::Inbound, stream),
```

Comment:
Won't buffering the inbound stream affect the rtt that the remote measures?
Since we do the buffering after the protocol negotiation succeeded, I think the remote will already have send a `Connect` message and start measuring the rtt.
So a subsequent holepunching attempt would be likely to fail because the timing is off?